python/shotgun_model/shotgun_query_model.py

```diff
@@ -0,0 +1,979 @@
+
+        :param node: :class:`~PySide.QtGui.QStandardItem` tree node
+        """
+
+        # depth first traversal
+        for index in xrange(node.rowCount()):
+            child_node = node.child(index)
+            self._do_depth_first_tree_deletion(child_node)
+
+        # delete the child leaves
+        for index in range(node.rowCount())[::-1]:
+            node.removeRow(index)
+
+    def _get_all_item_unique_ids(self):
+        """
+        Conveneince method. Returns the unique IDs of all items in the model.
+
+        The unique IDs correspond to the field defined by
+        ``SG_DATA_UNIQUE_ID_FIELD``
+
+        :return: A list of uniqe ids for all items in the model.
+        :rtype: ``list``
+        """
+
+        return self._tree_data.keys()
+
+    def _get_item_by_unique_id(self, uid):
+        """
+        Convenience method. Returns an item given a unique ID.
+
+        The unique ``uid`` corresponds to the field defined by
+        ``SG_DATA_UNIQUE_ID_FIELD``
+
+        :param id: The unique id for an item in the model.
+
+        :return: An item corresponding to the supplied uniqueid
+        :rtype: :class:`~PySide.QtGui.QStandardItem`
+        """
+
+        if uid not in self._tree_data:
+            return None
+
+        return self._tree_data[uid]
+
+    def _load_cached_data(self):
+        """
+        Convenience wrapper from loading cached data from disk.
+
+        Handles logging cache load attempts/failures.
+
+        :returns: ``True`` if data loaded from disk, ``False`` otherwise.
+        """
+
+        # warn if the cache file does not exist
+        if not self._cache_path or not os.path.exists(self._cache_path):
+            logger.debug(
+                "Data cache file does not exist on disk.\n"
+                "Looking here: %s" % (self._cache_path,)
+            )
+            return False
+
+        logger.debug(
+            "Now attempting cached data load from: %s ..." %
+            (self._cache_path,)
+        )
+
+        try:
+            time_before = time.time()
+            num_items = self._load_from_disk()
+            time_diff = (time.time() - time_before)
+            logger.debug(
+                "Loading finished! Loaded %s items in %4fs" %
+                (num_items, time_diff)
+            )
+            self.cache_loaded.emit()
+            return True
+        except Exception, e:
+            logger.debug(
+                "Couldn't load cache data from disk.\n"
+                " Will proceed with full SG load.\n"
+                "Error reported: %s" % (e,)
+            )
+            return False
+
+    def _sg_clean_data(self, sg_data):
+        """
+        Recursively clean the supplied SG data for use by clients.
+
+        This method currently handles:
+
+            - Converting datetime objects to universal time stamps.
+
+        :param sg_data:
+        :return:
+        """
+
+        # QT is struggling to handle the special timezone class that the
```

Comment:
yes pls remove it! I think we should probably do the same for the shotgun utils - could you take a look and potentially create a ticket?